pkg/core/transactor/transactor.go

```diff
@@ -73,6 +75,24 @@ func New(eb *eventbus.EventBus, rb *rpcbus.RPCBus, db database.DB, srv *grpc.Ser
 	return t, nil
 }
 
+func (t *Transactor) getSyncProgress() (*node.SyncProgressResponse, error) {
```

Comment:
Yeah, why would it go to other nodes?

Reply:
I wanted to ensure that we can use the grpc connection for internal communication (between components).

Reply:
For component-to-component communication we already use rpcbus, eventBus and pure callbacks. If @autholykos is OK with introducing  `grpc` messages for internal messaging, it's OK for me too

Reply:
Ah, I understand now. Well, it is a valid point, I opted to use existing infrastructure to get the information, but if we'd rather restrict inter-component communication to one of the mentioned options then I'm happy to change it.

But yes, we can use GRPC for internal communication as well, as demonstrated in the test harness :)

Reply:
Not sure what you meant by `demonstrated in the test harness`. Test-harness is a separate process which needs IPC to communicate with the network nodes.

Reply:
I mean that the new GRPC code is being ran in the test-harness, as it sends transactions for some of its tests.

Reply:
Test-harness is a separate process which needs IPC to communicate with the network nodes.

Reply:
I don't understand the need to argue semantics over something which is seemingly hard to misconstrue - the test harness runs tests in which nodes send transactions, meaning this internal GRPC logic (as introduced in this PR) is called, so this logic is verifiably working, to adress your point above.

Reply:
I just wanted to avoid misunderstanding. There is no thing like `- the test harness runs tests in which nodes send transactions`. Test-harness runs as a simple GRPC client (external component) that sends Txs to nodes.

That said, we have not used anywhere GRPC service for communication between internal components.

Reply:
I think that the `test-harness` brought the conversation a bit off-topic. It is not intended for deployment on production, so I don't quite care what it uses to accomplish its goals (i.e. help identify problems early by running different scenarios). I think the concern really is using different methodology of communications when there isn't a strong reason to do so. 

At the moment the way for processes to communicate are:
- `RPCBus` for synchronous communication among light-threaded components
- `EventBus`for internal asynchronous message passing
- `GRPC` for interprocess synchronous communication
- `websocket` for interprocess asynchronous communication

This would suggest that using GRPC for communication among light-threded components would break consistency. While this is fundamentally true, this raises an interesting thought, i.e. to unify our IPC method and use just one. This would have the obvious advantage of consistency and maintainability. 

Unfortunately the `RPCBus` does not support crossing process boundaries, it would make an unsuitable candidate ofr such a unification, so this would leave GRPC.

So the real question we need to ask yourself are 
1. *Is there any advantage in keeping the RPCBus alongside GRPC?*
2. *Do these advantages justify keeping two protocols rather than one(albeit for slightly different contexts)*

If the answer is *NO* to any of the former question, it would make sense to keep the GRPC interface for `GetSyncProgress` and consider the other `RPCBus` calls as _tech-debt_ to be solved by slowly migrating to GRPC   in the future.

If the answer is *YES* to both question, then `GetSyncProgress` needs to be migrated to `RPCBus`. In this case, this would be also be consider _tech-debt_ and should be tackled on a separate issue.

So, for now, the GRPC method can stay. However, the aforementioned discussion should highlight what _tech-debt issue_ we wanna tackle in the near future.